### ISSUE TITLE
Cannot build mbgmt anymore on Win

### ISSUE THREAD

**Issue by joa-quim** [GenericMappingTools/gmt#3090]
Either #2929 or a subsequent change had broken it. Now I can only add the MB to the ``gmtsupplements``. I still manage to create a ``mbgmt_wXX.dll`` but it has nothing exported in it.

**Comment by seisman** [comment]
Any instructions to build mbgmt on macOS?

**Comment by joa-quim** [comment]
There is but it's not the same thing. My build on Windows is unique (in the sense that I'm the only one crazy enough to try to maintain it without any support). But it doesn't really matter if its MB or any other supplement that depend in other external libs that the GMT ones (and its dependencies).

**Comment by PaulWessel** [comment]
I should try to build my extra supplement [GSFLM](http://www.soest.hawaii.edu/PT/GSFML/) for handling data by the same name.  I think the initial idea with the CMake changes was to make it simple for external supplements to "plug in" to the GMT build but produce separate file.so or file.dll that can be placed in the plugin, rather than go into supplements.so.

**Comment by joa-quim** [comment]
Yes, but things turned out to be the other way around.

**Comment by PaulWessel** [comment]
I will try to have a look.  I did get my GSFML cmakelist.txt file to build gsfml.so but it almost duplicated all of the gmt one.  I will give it a try but not sure when.  Too many shits to sort out first, including grd2kml.  At least grdvector is now on solid grounds.

**Comment by seisman** [comment]
After #3092 and #3111, it should be straightforward to add a new supplement package to GMT. Anyone can build a src subdirectory into a supplement library, as long as there is a CMakeLists.txt in the subdirectory which contains some CMake settings.

@joa-quim For your mbgmt package, you may try a CMakeLists.txt like this (you may need to change some details since I know nothing about mbgmt). Note that you don't need to set EXTRA_MODULES_SUPPL and EXTRA_INCLUDE_NEWSUPPL anymore.
```
set (SUPPL_NAME mb)
set (SUPPL_LIB_NAME mb)
set (SUPPL_PROGS_SRCS mbcontour.c mbswath.c mbgrdtiff.c mbclean_j.c mbgrid_j.c mbimport.c
	mbgetdata.c mbflags.c mbset.c mbinfo.c mbareaclean.c gmtmbgrid.c mbprocess.c)
set (SUPPL_LIB_SRCS ${SUPPL_PROGS_SRCS} mbgmt_module.c)
set (SUPPL_EXTRA_INCLUDES ${ONCRPCINC} ${GETOPTINC} ${PROJINCDIR} ${MBINC})
set (SUPPL_EXTRA_LIBS ${PROJLIB} ${ONCRPCLIB} ${GETOPTLIB} ${MBIOLIB} ${MBAUXLIB})
```

**Comment by joa-quim** [comment]
Ok, thanks. It works but lost the ability to set a DLL rename. I used to do it in the module's CMakeList.txt but it now errors. Probably too soon. So I was forced to add this to the ``src/CMakeList.txt``
```
if (WIN32 AND MB_DLL_RENAME)
	set_target_properties (${_suppl_lib_name} PROPERTIES RUNTIME_OUTPUT_NAME ${MB_DLL_RENAME})
endif (WIN32 AND MB_DLL_RENAME)
```
but that is not generic enough

**Comment by seisman** [comment]
> It works but lost the ability to set a DLL rename. 

Yes, it's a known issue for me. Probably the best way is to allow a "SUPPL_DLL_RENAME" setting in src/mb/CMakeLists.txt. I'll see how it works.

**Comment by seisman** [comment]
I think the issue was fixed by #3125. 

@joa-quim OK to close the issue?

**Comment by joa-quim** [comment]
Yes, and thanks for your work on this.
